agent/app/agent.go

```diff
@@ -216,6 +218,10 @@ func newAgent(blackholeEC2Metadata bool, acceptInsecureCert *bool) (agent, error
 	}, nil
 }
 
+func (agent *ecsAgent) getConfig() *config.Config {
```

Comment:
The config is an attribute of `ecsAgent` struct, initialized here: https://github.com/aws/amazon-ecs-agent/blob/master/agent/app/agent.go#L143

But I actually use the config outside (one level up), here: https://github.com/aws/amazon-ecs-agent/pull/3001/files#diff-13d2f327a704753f102c6d2a64d5ea2f4a2a03dfdb37abe733f21e9f4c291697R76 

The reference to cfg is not accessible where I need it because the `newAgent` method returns an `agent` [interface](https://github.com/aws/amazon-ecs-agent/blob/master/agent/app/agent.go#L93), not the actual `ecsAgent` struct. Therefore I had to add `getConfig` to the `agent` interface in order to have access to the config.